pkg/lorry/plugin/proto/db_plugin.proto

```diff
@@ -0,0 +1,255 @@
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message AccountProvisionRequest {
+  DBInfo db_info = 1;
+
+  // the user name of the new account.
+  string user_name = 2;
+
+  // the password of the new account.
+  string password = 3 [(kb_secret) = true];
+
+  // the role of the new account.
+  string role = 4;
+
+  // Common metadata property for extention
+}
+
+message AccountProvisionResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message SwitchoverRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Name of the old primary replica to demote.
+  string primary = 4;
+
+  // The Name of the new primary replica.
+  string candidate = 5;
+
+  // Force the switchover to happen even if the primary is not ready.
+  bool force = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message SwitchoverResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+// DBPlugin service provides APIs to kubeblocks operator to exec component operation.
+service DBPlugin {
+  rpc GetPluginInfo(GetPluginInfoRequest) returns (GetPluginInfoResponse) {}
+
+  // IsDBReady defines the mechanism to probe the readiness of the database.
+  rpc IsDBReady(IsDBReadyRequest) returns (IsDBReadyResponse) {}
+
+  // GetRole defines the mechanism to probe the role of replicas. The return role
+  // must be one of the names defined in the componentdefinition roles. 
+  rpc GetRole(GetRoleRequest) returns (GetRoleResponse) {}
+
+  // MemberJoin defines how to add a new replica to the replication group. 
+  // This action is typically invoked when a new replica needs to be added, 
+  // such as during scale-out. It may involve updating configuration, 
+  // notifying other members, and ensuring data consistency.
+  rpc JoinMember(JoinMemberRequest) returns (JoinMemberResponse) {}
+
+  // MemberLeave defines how to remove a replica from the replication group. 
+  // This action is typically invoked when a replica needs to be removed, 
+  // such as during scale-in. It may involve configuration updates and notifying
+  // other members about the departure,
+  rpc LeaveMember(LeaveMemberRequest) returns (LeaveMemberResponse) {}
+
+  // Readonly defines how to set a replica service as read-only.
+  rpc Readonly(ReadonlyRequest) returns (ReadonlyResponse) {}
+
+  // Readwrite defines how to set a replica service as read-write.
+  rpc Readwrite(ReadwriteRequest) returns (ReadwriteResponse) {}
+
+  // AccountProvision Defines the procedure to generate a new database account.
+  rpc AccountProvision(AccountProvisionRequest) returns (AccountProvisionResponse) {}
+
+  // Switchover defines the procedure to switch roles of the primary and candidate.
```

Comment:
// Switchover defines the procedure to switch replica roles in a primary-secondary HA DBEngine by promoting the secondary to primary and demoting the current primary to secondary.